05-UI-ReactJS/App.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
+import './App.css';
+import MovieCreator from './MovieCreator/MovieCreator';
+import MovieList from './MovieList/MovieList';
+
+
+class App extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      movies: [ //Default movies
+        {title: "Toy Story", year: 1995, mins: 90},
+        {title: "The Hobbit", year: 2014, mins: 150},
+        {title: "Star Wars", year: 1987, mins: 110},
+        {title: "Iron Man", year: 2012, mins: 120}
+      ],
+      showMovies: true, //allows to show/hide movies with toggleMoviesHandler()
+      newMovie: null
+    }
+  }
+
+
+  toggleMoviesHandler = ()=>{ //shows or hides list of movies
+    const doesShow = this.state.showMovies;
+    this.setState({showMovies: !doesShow});
+  }
+
+  deleteMovieHandler = (movieIndex)=>{     //deletes a movie from state.movies
+    const movies = [...this.state.movies]; // makes copy of state.movies
+    movies.splice(movieIndex,1);          //Deletes selected movie
+    this.setState({movies: movies});      //Updates state
+  }
+
+  newMovieTitle = (event)=>{ ///changes title of state.newMovie for input value
+    const movie = {...this.state.newMovie};
+    movie.title = event.target.value;
+    this.setState({newMovie:movie});
+  }
+
+  newMovieYear = (event)=>{ ///changes year of state.newMovie for input value
+    const movie = {...this.state.newMovie};
+    movie.year = event.target.value;
+    this.setState({newMovie:movie});
+  }
+
+  newMovieMins = (event)=>{///changes mins of state.newMovie for input value
+    const movie = {...this.state.newMovie};
+    movie.mins = event.target.value;
+    this.setState({newMovie:movie});
+  }
+
+
+  addMovieHandler = ()=>{ //adds newMovie to state.movies
+    const movies = [...this.state.movies]; //Makes a copy of state.movies
```

Comment:
Acá podrías haber hecho 
`this.setState({movies: [...this.state.movies, newMovie]});`